### ISSUE TITLE
Support cleaning up wrappers automagically

### ISSUE THREAD

**Issue by ghost** [vuejs/vue-test-utils#1236]
### What problem does this feature solve?

Currently it is necessary to call `wrapper.destroy()` / `wrapper.vm.$destroy()` manually. Forgetting to do so may result in difficult to track down side effects and performance degradation (especially when not mocking Vuex).

### What does the proposed API look like?

Introduce a new `enableAutoDestroy()`  helper which accepts a hook function such as [`afterEach()`](https://jestjs.io/docs/en/api#aftereachfn-timeout) to call `wrapper.destroy()` as callback.

**Comment by winniehell** [comment]
@eddyerburgh I am interested in implementing this. Would you accept a pull request? :smiley:

**Comment by eddyerburgh** [comment]
Yep 😄

**Comment by winniehell** [comment]
As I realized from @souldzin's comment on the pull request (https://github.com/vuejs/vue-test-utils/pull/1240#issuecomment-494851349), the API proposed here does not work. For a hook like `afterEach` we want to make sure it is only called exactly once. I'm not sure how we would ensure that when passing it as an config option.

Instead we could provide an `enableAutoDestroy()` function that accepts the hook function as parameter. We could then rely on that function being called only once (in test setup).

**Comment by winniehell** [comment]
I have updated the description now with the new suggested API and will create a new pull request tomorrow.
